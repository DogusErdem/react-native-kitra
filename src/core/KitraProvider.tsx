import type { ComponentType, ForwardedRef } from 'react';
import React from 'react';
import { GestureHandlerRootView } from 'react-native-gesture-handler';
import Animated from 'react-native-reanimated';
import { StyleProp, ViewStyle } from 'react-native';

import { ThemeProvider } from './theme/theme';
import { TypographyContext, TypographyProvider } from './typography/typography';
import { NotificationContextType, NotificationProvider } from './notification/notification';

const Feather = require('react-native-vector-icons/Feather').default;
const Ionicons = require('react-native-vector-icons/Ionicons').default;
const AntDesign = require('react-native-vector-icons/AntDesign').default;

Animated.addWhitelistedNativeProps({ text: true });
const messageTypes = (theme:any) => ({
  SUCCESS: {
    backgroundColor: theme.colors.status.successLight,
    icon: <Feather name="check" size={25} color={theme.colors.status.success} />,
  },
  INFO: {
    backgroundColor: theme.colors.system.primary15,
    icon: <Ionicons name="information-circle-outline" size={25} color={theme.colors.system.primary} />,
  },
  DANGER: {
    backgroundColor: theme.colors.status.errorLight,
    icon: <AntDesign name="warning" size={25} color={theme.colors.status.error} />,
  },
  WARNING: {
    backgroundColor: theme.colors.status.warningLight,
    icon: <AntDesign name="warning" size={25} color={theme.colors.status.warning} />,
  },
});

type KitraProviderType= {
  children?:any,
  limit?:number,
  notificationRef?:React.Ref<NotificationContextType>,
  messageType?:(theme?:any, colorScheme?:'dark' | 'light') => {[key:string]: {
      backgroundColor:string,
    icon?:React.ReactNode
    onPress?:()=>void
  }},
  notificationCcontainerStyle?:StyleProp<ViewStyle>,
  customView?:({ header, type, message, theme }:
    {header:string, type:string, message:string, theme:any})=>React.ReactNode}

export const KitraProvider =
({ children, messageType = messageTypes, notificationCcontainerStyle, limit, customView,
  notificationRef }: KitraProviderType) => (
    <GestureHandlerRootView style={{ flex: 1 }}>
      <ThemeProvider>
        <TypographyProvider>
          <NotificationProvider
            messageType={theme => (messageType ? messageType(theme) : theme)}
            notificationCcontainerStyle={notificationCcontainerStyle}
            limit={limit}
            customView={item => customView?.(item)}
            ref={notificationRef}
          >
            {children}
          </NotificationProvider>
        </TypographyProvider>
      </ThemeProvider>
    </GestureHandlerRootView>
);

export function applyDefaults<T>(Component: ComponentType<T>) {
  return React.forwardRef((props: T, ref: ForwardedRef<any>) => (
    <TypographyContext.Consumer>
      {typography =>
        <Component ref={ref} typography={typography?.typography} {...props} />
          }
    </TypographyContext.Consumer>
  ));
}
