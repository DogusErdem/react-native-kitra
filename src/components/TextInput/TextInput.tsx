import React, { useEffect, useRef, useState } from 'react';
import { TextInput as RNTextInput,
  View, TextInputProps as RNTextInputProps, TouchableOpacity, StyleSheet, Text,
  TextStyle } from 'react-native';
import Animated,
{ interpolate, interpolateColor, useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated';
import useComponentTheme from '../../core/hooks/useComponentTheme';
import type { FCCWD, TextInputProps } from '../../types';
import { applyDefaults } from '../../core/KitraProvider';

const AnimatedTextInput = Animated.createAnimatedComponent(RNTextInput);
const TextInput: FCCWD<TextInputProps & RNTextInputProps> = (
  { inputStyle,
    editable = true,
    size = 'medium',
    bottomLabel = '',
    variant = 'filled',
    labelContainerStyle,
    containerStyle,
    count = false,
    onChangeText,
    label,
    inputContainerStyle,
    labelStyle,
    bottomLabelStyle,
    error = false,
    left,
    right,
    onFocus,
    onEndEditing,
    theme,
    typography,
    ...props },
) => {
  const [isFocused, setIsFocused] = useState(false);
  const inputRef = useRef<RNTextInput>(null);
  const [counts, setCounts] = useState(0);
  const [labelLayout, setLabelLayout] = useState({ width: 0, height: 0 });
  const textInputOffset = useSharedValue(props.defaultValue || props.placeholder || props.value ? 0 : 1);
  const { statusTheme, componentTheme } = useComponentTheme(
    theme,
    'textInput',
    editable ? (error ? 'error' : isFocused ? 'focused' : 'default') : 'disabled',
  );
  const { value: statusValue } = statusTheme;

  const fontStyles =
  {
    large: typography?.body.regular,
    medium: typography?.body.sregular,
    small: typography?.body.xsregular,
  };

  const sizeStyles =
  {
    large: {
      paddingHorizontal: 15,
      height: 51,
      paddingVertical: 17,
      lineHeight: 21,
    },
    medium: {
      paddingHorizontal: 12,
      height: 42,
      paddingVertical: 12,
      lineHeight: 18,
    },
    small: {
      paddingHorizontal: 10,
      height: 36,
      paddingVertical: 10,
      lineHeight: 15,
    },
  };

  const labelStyles =
  {
    large: {
      focused: {
        fontSize: 14,
        lineHeight: 16,
        fontFamily: 'Poppins-Regular',
      },
      default: {
        fontSize: 12,
        lineHeight: 16,
        fontFamily: 'Poppins-Regular',
      },
    },
    medium: {
      focused: {
        fontSize: 12,
        lineHeight: 16,
        fontFamily: 'Poppins-Regular',
      },
      default: {
        fontSize: 10,
        lineHeight: 14,
        fontFamily: 'Poppins-Regular',
      },
    },
    small: {
      focused: {
        fontSize: 10,
        lineHeight: 14,
        fontFamily: 'Poppins-Regular',
      },
      default: {
        fontSize: 8,
        lineHeight: 12,
        fontFamily: 'Poppins-Regular',
      },
    },
  };

  const labelPositionAnimation = useAnimatedStyle(() => {
    const topInterpolate = interpolate(
      textInputOffset.value,
      [0, 1],
      [-1, sizeStyles[size].paddingVertical],
    );

    return {
      top: topInterpolate,
    };
  });

  const borderAnimation = useAnimatedStyle(() => {
    const topInterpolate = interpolateColor(
      textInputOffset.value,
      [0, 1],
      [componentTheme.focused?.border || '',
        editable ? (error ? componentTheme.error?.border || '' : componentTheme.default?.border || '')
          : componentTheme.disabled?.border || ''],
    );

    return {
      borderColor: topInterpolate,
    };
  });

  const labelFontAnimation = useAnimatedStyle(() => {
    const fontSize = interpolate(
      textInputOffset.value,
      [0, 1],
      [labelStyles[size].default.fontSize,
        (labelStyle as TextStyle)?.fontSize || labelStyles[size].focused.fontSize],
    );
    const lineHeight = interpolate(
      textInputOffset.value,
      [0, 1],
      [labelStyles[size].default.lineHeight,
        (labelStyle as TextStyle)?.lineHeight || labelStyles[size].focused.lineHeight],
    );
    const color = interpolateColor(
      textInputOffset.value,
      [0, 1],
      // @ts-ignore
      [statusTheme.label, statusTheme.label],
    );
    const topInterpolate = interpolate(
      textInputOffset.value,
      [0, 1],
      [variant === 'filled' ? 2 : -labelStyles[size].focused.lineHeight + labelStyles[size].focused.lineHeight / 2,
        sizeStyles[size].paddingVertical],
    );
    const leftInterpolate = interpolate(
      textInputOffset.value,
      [0, 1],
      [variant === 'outlined' ? 4 : 0, 0],
    );
    return {
      fontSize,
      lineHeight,
      color,
      top: topInterpolate,
      left: leftInterpolate,
    };
  });

  useEffect(() => {
    if (props.value?.length) {
      setCounts(props.value.length);
    } else if (props.defaultValue?.length) {
      setCounts(props.defaultValue?.length);
    }
  }, [props.value, props.defaultValue]);

  const onFocusInput = () => {
    textInputOffset.value = withTiming(0);
    setIsFocused(true);
  };

  const onEndEditingInput = () => {
    if (counts === 0 && !props.placeholder) {
      textInputOffset.value = (counts === 0) ? withTiming(1) : withTiming(0);
    }
    setIsFocused(false);
  };

  useEffect(() => {
    if (!props?.value && !isFocused) {
      textInputOffset.value = withTiming(1);
      setIsFocused(false);
      setCounts(0);
    } else {
      textInputOffset.value = withTiming(0);
      setIsFocused(true);
    }
  }, [props?.value]);

  return (
    <View testID="inputContainer" style={[containerStyle]}>
      <Animated.View
        testID="inputInnerContainer"
        style={[{
          borderRadius: 5,
          borderWidth: editable ? 1 : 0,
          height: sizeStyles[size].height,
          maxHeight: sizeStyles[size].height,
        },
        // @ts-ignore
        borderAnimation, inputContainerStyle, { backgroundColor: statusTheme.background }]}
      >
        <View
          testID="inputInside"
          style={{ flex: 1,
            flexDirection: 'row',
            height: sizeStyles[size].height,
            paddingHorizontal: sizeStyles[size].paddingHorizontal }}
        >
          {typeof left === 'function' && (
            <View style={{ alignSelf: 'center',
              marginRight: 5 }}
            >
              { left?.(isFocused)}
            </View>
          )}

          <View style={{ flex: 1, flexDirection: 'row' }}>
            {variant === 'outlined' && (
            <Animated.View
              testID="outlinedContainer"
              style={[{ position: 'absolute',
                width: labelLayout.width + 8,
                height: 2,
                zIndex: 100 }, labelPositionAnimation, { backgroundColor: statusTheme.background }]}
            />
            )}
            <AnimatedTextInput
              testID="input"
              ref={inputRef}
              editable={editable}
              style={[{
                marginTop: variant === 'filled' ? fontStyles[size].lineHeight / 2 : 0,
                fontSize: fontStyles[size].fontSize,
                fontFamily: labelStyles[size].default.fontFamily,
                lineHeight: sizeStyles[size].lineHeight,
                flexDirection: 'row',
                flexGrow: 1,
              }, inputStyle, { backgroundColor: statusTheme.background, color: statusValue }]}
              onChangeText={event => { setCounts(event?.length || 0); onChangeText && onChangeText(event); }}
              onFocus={x => { onFocusInput(); onFocus?.(x); }}
              onEndEditing={x => { onEndEditingInput(); onEndEditing?.(x); }}
              {...props}
            />
            {label ? (
              <View
                testID="labelContainer"
                style={[
                  {
                    position: 'absolute',
                    zIndex: 101,
                  },
                  labelContainerStyle,
                ]}
              >
                <TouchableOpacity
                  onPress={() => inputRef.current?.focus()}
                  activeOpacity={0.9}
                  onLayout={event => setLabelLayout({ width: event.nativeEvent.layout.width,
                    height: event.nativeEvent.layout.height })}
                >
                  <Animated.Text
                    testID="inputLabel"
                    style={[{ fontFamily: labelStyles[size].default.fontFamily },
                      labelStyle, labelFontAnimation, { color: statusTheme.label }]}
                  >
                    {label}
                  </Animated.Text>
                </TouchableOpacity>
              </View>
            ) : null
            }
          </View>
          {typeof right === 'function' && (
          <View style={{ alignSelf: 'center' }}>
            {right?.(isFocused)}
          </View>
          )}

        </View>
      </Animated.View>

      <View style={[styles.helperContainer, { flexDirection: 'row',
        display: (error || !!bottomLabel || count) ? 'flex' : 'none',
        alignSelf: 'stretch' }]}
      >
        <Text
          testID="inputBottomLabel"
          style={[labelStyles[size].default, bottomLabelStyle,
            { color: statusTheme.bottomLabel }]}
        >
          {bottomLabel}
        </Text>
        {count ? (
          <Text
            testID="inputCountLabel"
            style={[labelStyles[size].default, bottomLabelStyle,
              { color: statusTheme.countLabel }]}
          >
            {`${counts}/${props.maxLength}`}
          </Text>
        ) : null}
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  inputContainer: {
    alignItems: 'center',
  },
  helperContainer: {
    flexDirection: 'row',
    paddingHorizontal: 5,
    justifyContent: 'space-between',
  },

});
export default applyDefaults(TextInput);
