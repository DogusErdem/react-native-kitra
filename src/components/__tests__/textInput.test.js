import { fireEvent, render } from '@testing-library/react-native';
import { KitraProvider } from '../../core/KitraProvider';
import TextInput from '../TextInput/TextInput';

/* eslint-disable no-undef */
const renderWithContext = children => render(<KitraProvider>{children}</KitraProvider>);

describe('Text Input', () => {
  it('default render', () => {
    const tree = renderWithContext(
      <TextInput label="Label" />,
    );
    expect(tree).toMatchSnapshot();
  });
  describe('render with different size', () => {
    it('render with small size', () => {
      const { getByTestId } = renderWithContext(
        <TextInput
          size="small"
          label="Label"
          bottomLabel="Test"
          count
          maxLength={20}
        />,
      );
      const inputInnerContainer = getByTestId('inputInnerContainer');
      const inputInside = getByTestId('inputInside');
      const input = getByTestId('input');
      const inputLabel = getByTestId('inputLabel');
      const inputBottomLabel = getByTestId('inputBottomLabel');
      const inputCountLabel = getByTestId('inputCountLabel');
      expect(input).toHaveStyle({ marginTop: 7, fontSize: 10, fontFamily: 'Poppins-Regular', lineHeight: 15 });
      expect(inputInnerContainer).toHaveStyle({ height: 36, maxHeight: 36 });
      expect(inputInside).toHaveStyle({ height: 36, paddingHorizontal: 10 });
      expect(inputLabel).toHaveStyle({ fontFamily: 'Poppins-Regular' });
      expect(inputBottomLabel).toHaveStyle({ fontFamily: 'Poppins-Regular', fontSize: 8, lineHeight: 12 });
      expect(inputCountLabel).toHaveStyle({ fontFamily: 'Poppins-Regular', fontSize: 8, lineHeight: 12 });
    });
    it('render with medium size', () => {
      const { getByTestId } = renderWithContext(
        <TextInput
          size="medium"
          label="Label"
          bottomLabel="Test"
          count
          maxLength={20}
        />,
      );
      const inputInnerContainer = getByTestId('inputInnerContainer');
      const inputInside = getByTestId('inputInside');
      const input = getByTestId('input');
      const inputLabel = getByTestId('inputLabel');
      const inputBottomLabel = getByTestId('inputBottomLabel');
      const inputCountLabel = getByTestId('inputCountLabel');
      expect(inputInnerContainer).toHaveStyle({ height: 42, maxHeight: 42 });
      expect(inputInside).toHaveStyle({ height: 42, paddingHorizontal: 12 });
      expect(input).toHaveStyle({ marginTop: 8, fontSize: 12, fontFamily: 'Poppins-Regular', lineHeight: 18 });
      expect(inputLabel).toHaveStyle({ fontFamily: 'Poppins-Regular' });
      expect(inputBottomLabel).toHaveStyle({ fontFamily: 'Poppins-Regular', fontSize: 10, lineHeight: 14 });
      expect(inputCountLabel).toHaveStyle({ fontFamily: 'Poppins-Regular', fontSize: 10, lineHeight: 14 });
    });
    it('render with large size', () => {
      const { getByTestId } = renderWithContext(
        <TextInput
          size="large"
          label="Label"
          bottomLabel="Test"
          count
          maxLength={20}
        />,
      );
      const inputInnerContainer = getByTestId('inputInnerContainer');
      const inputInside = getByTestId('inputInside');
      const input = getByTestId('input');
      const inputLabel = getByTestId('inputLabel');
      const inputBottomLabel = getByTestId('inputBottomLabel');
      const inputCountLabel = getByTestId('inputCountLabel');

      expect(inputInnerContainer).toHaveStyle({ height: 51, maxHeight: 51 });
      expect(inputInside).toHaveStyle({ height: 51, paddingHorizontal: 15 });
      expect(input).toHaveStyle({ marginTop: 8, fontSize: 14, fontFamily: 'Poppins-Regular', lineHeight: 21 });
      expect(inputLabel).toHaveStyle({ fontFamily: 'Poppins-Regular' });
      expect(inputBottomLabel).toHaveStyle({ fontFamily: 'Poppins-Regular', fontSize: 12, lineHeight: 16 });
      expect(inputCountLabel).toHaveStyle({ fontFamily: 'Poppins-Regular', fontSize: 12, lineHeight: 16 });
    });
  });
  describe('render with onChangeText prop', () => {
    const onChangeText = jest.fn();
    const { getByTestId } = renderWithContext(
      <TextInput
        label="Label"
        onChangeText={onChangeText}
      />,
    );
    fireEvent.changeText(getByTestId('input'), 'Hello World');
    expect(onChangeText).toHaveBeenCalledWith('Hello World');
    expect(onChangeText).toHaveBeenCalledTimes(1);
  });
  describe('render with onFocus and onEndEditing props', () => {
    const onFocus = jest.fn();
    const onEndEditing = jest.fn();

    const { getByPlaceholderText, getByTestId } = renderWithContext(
      <TextInput label="Label" onFocus={onFocus} onEndEditing={onEndEditing} />,
    );
    const input = getByTestId('input');
    fireEvent(input, 'focus');
    expect(onFocus).toHaveBeenCalledTimes(1);

    fireEvent(input, 'endEditing');
    expect(onEndEditing).toHaveBeenCalledTimes(1);
  });
  describe('render with left right props', () => {
    const left = jest.fn();
    const right = jest.fn();
    const { getByTestId } = renderWithContext(
      <TextInput
        left={left}
        right={right}
      />,
    );
    const input = getByTestId('input');
    expect(left).toHaveBeenCalledWith(false);
    expect(right).toHaveBeenCalledWith(false);
    fireEvent(input, 'focus');
    expect(left).toHaveBeenCalledWith(true);
    expect(right).toHaveBeenCalledWith(true);
  });
  describe('render with custom theme', () => {
    it('render with default theme', () => {
      const { getByTestId } = renderWithContext(
        <TextInput
          label="Label"
          bottomLabel="Bottom Label"
          count
          variant="outlined"
          maxLength={10}
          theme={{ default: {
            background: '#F6F9FF',
            border: '#195CEF',
            bottomLabel: '#195CEF',
            countLabel: '#195CEF',
            label: '#195CEF',
            value: '#195CEF',
          } }}
        />,
      );
      const inputInnerContainer = getByTestId('inputInnerContainer');// background
      const outlinedContainer = getByTestId('outlinedContainer');// background
      const input = getByTestId('input');// background
      const inputLabel = getByTestId('inputLabel');// label
      const inputBottomLabel = getByTestId('inputBottomLabel');// bottomLabel
      const inputCountLabel = getByTestId('inputCountLabel');// inputCountLabel

      expect(inputInnerContainer).toHaveStyle({ backgroundColor: '#F6F9FF' });
      expect(outlinedContainer).toHaveStyle({ backgroundColor: '#F6F9FF' });
      expect(input).toHaveStyle({ backgroundColor: '#F6F9FF' });
      expect(inputLabel).toHaveStyle({ color: '#195CEF' });
      expect(inputBottomLabel).toHaveStyle({ color: '#195CEF' });
      expect(inputCountLabel).toHaveStyle({ color: '#195CEF' });
    });
    it('render with focus theme', () => {
      const { getByTestId } = renderWithContext(
        <TextInput
          label="Label"
          bottomLabel="Bottom Label"
          count
          variant="outlined"
          maxLength={10}
          theme={{ focused: {
            background: 'white',
            border: 'grey',
            bottomLabel: '#195CEF',
            countLabel: 'red',
            label: 'black',
            value: '#195CEF',
          } }}
        />,
      );
      const inputInnerContainer = getByTestId('inputInnerContainer');// background
      const outlinedContainer = getByTestId('outlinedContainer');// background
      const input = getByTestId('input');// background
      const inputLabel = getByTestId('inputLabel');// label
      const inputBottomLabel = getByTestId('inputBottomLabel');// bottomLabel
      const inputCountLabel = getByTestId('inputCountLabel');// inputCountLabel
      fireEvent(input, 'focus');
      expect(inputInnerContainer).toHaveStyle({ backgroundColor: 'white' });
      expect(outlinedContainer).toHaveStyle({ backgroundColor: 'white' });
      expect(input).toHaveStyle({ backgroundColor: 'white' });
      expect(inputLabel).toHaveStyle({ color: 'black' });
      expect(inputBottomLabel).toHaveStyle({ color: '#195CEF' });
      expect(inputCountLabel).toHaveStyle({ color: 'red' });
    });
    it('render with error theme', () => {
      const { getByTestId } = renderWithContext(
        <TextInput
          label="Label"
          bottomLabel="Bottom Label"
          count
          variant="outlined"
          error
          maxLength={10}
          theme={{ error: {
            background: 'white',
            border: 'red',
            bottomLabel: 'red',
            countLabel: 'red',
            label: 'red',
            value: 'red',
          } }}
        />,
      );
      const inputInnerContainer = getByTestId('inputInnerContainer');// background
      const outlinedContainer = getByTestId('outlinedContainer');// background
      const input = getByTestId('input');// background
      const inputLabel = getByTestId('inputLabel');// label
      const inputBottomLabel = getByTestId('inputBottomLabel');// bottomLabel
      const inputCountLabel = getByTestId('inputCountLabel');// inputCountLabel
      expect(inputInnerContainer).toHaveStyle({ backgroundColor: 'white' });
      expect(outlinedContainer).toHaveStyle({ backgroundColor: 'white' });
      expect(input).toHaveStyle({ backgroundColor: 'white' });
      expect(inputLabel).toHaveStyle({ color: 'red' });
      expect(inputBottomLabel).toHaveStyle({ color: 'red' });
      expect(inputCountLabel).toHaveStyle({ color: 'red' });
    });
    it('render with error theme', () => {
      const { getByTestId } = renderWithContext(
        <TextInput
          label="Label"
          bottomLabel="Bottom Label"
          count
          editable={false}
          variant="outlined"
          maxLength={10}
          theme={{ disabled: {
            background: 'grey',
            border: 'grey',
            bottomLabel: 'grey',
            countLabel: 'grey',
            label: 'white',
            value: 'white',
          } }}
        />,
      );
      const inputInnerContainer = getByTestId('inputInnerContainer');// background
      const outlinedContainer = getByTestId('outlinedContainer');// background
      const input = getByTestId('input');// background
      const inputLabel = getByTestId('inputLabel');// label
      const inputBottomLabel = getByTestId('inputBottomLabel');// bottomLabel
      const inputCountLabel = getByTestId('inputCountLabel');// inputCountLabel
      expect(inputInnerContainer).toHaveStyle({ backgroundColor: 'grey' });
      expect(outlinedContainer).toHaveStyle({ backgroundColor: 'grey' });
      expect(input).toHaveStyle({ backgroundColor: 'grey' });
      expect(inputLabel).toHaveStyle({ color: 'white' });
      expect(inputBottomLabel).toHaveStyle({ color: 'grey' });
      expect(inputCountLabel).toHaveStyle({ color: 'grey' });
    });
  });
  describe('render with count,label and bottomLabel props', () => {
    const { getByTestId, getByText } = renderWithContext(
      <TextInput
        label="Label"
        count
        maxLength={10}
        bottomLabel="Bottom Label"
      />,
    );
    fireEvent.changeText(getByTestId('input'), 'HelloWorld');
    expect(getByText('10/10')).toBeTruthy();
    expect(getByText('Bottom Label')).toBeTruthy();
    expect(getByText('Label')).toBeTruthy();
  });
  describe('render with custom styles', () => {
    const { getByTestId } = renderWithContext(
      <TextInput
        label="Label"
        bottomLabel="Bottom Label"
        count
        inputStyle={{ fontSize: 12, fontWeight: 'bold', lineHeight: 11 }}// input
        labelStyle={{ fontWeight: '500' }}// inputLabel
        containerStyle={{ marginTop: 10, paddingHorizontal: 13, paddingVertical: 3 }}// inputContainer
        bottomLabelStyle={{ fontSize: 8, fontWeight: '500', lineHeight: 10 }}// inputBottomLabel , inputCountLabel
        inputContainerStyle={{ marginTop: 6, paddingHorizontal: 9, paddingVertical: 5 }}// inputInnerContainer
        labelContainerStyle={{ marginTop: 3, paddingHorizontal: 6, paddingVertical: 2 }} // labelContainer
      />,
    );
    const inputInnerContainer = getByTestId('inputInnerContainer');
    const inputContainer = getByTestId('inputContainer');
    const labelContainer = getByTestId('labelContainer');
    const input = getByTestId('input');
    const inputLabel = getByTestId('inputLabel');
    const inputBottomLabel = getByTestId('inputBottomLabel');
    const inputCountLabel = getByTestId('inputCountLabel');

    expect(input).toHaveStyle({ fontSize: 12, fontWeight: 'bold', lineHeight: 11 });
    expect(inputLabel).toHaveStyle({ fontWeight: '500' });
    expect(inputContainer).toHaveStyle({ marginTop: 10, paddingHorizontal: 13, paddingVertical: 3 });
    expect(inputBottomLabel).toHaveStyle({ fontSize: 8, fontWeight: '500', lineHeight: 10 });
    expect(inputCountLabel).toHaveStyle({ fontSize: 8, fontWeight: '500', lineHeight: 10 });
    expect(inputInnerContainer).toHaveStyle({ marginTop: 6, paddingHorizontal: 9, paddingVertical: 5 });
    expect(labelContainer).toHaveStyle({ marginTop: 3, paddingHorizontal: 6, paddingVertical: 2 });
  });
});
