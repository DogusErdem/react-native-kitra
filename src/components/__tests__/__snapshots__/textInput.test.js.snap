// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input default render 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        undefined,
      ]
    }
    testID="inputContainer"
  >
    <View
      style={
        [
          {
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 42,
            "maxHeight": 42,
          },
          {
            "borderColor": undefined,
          },
          undefined,
          {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
      testID="inputInnerContainer"
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
            "height": 42,
            "paddingHorizontal": 12,
          }
        }
        testID="inputInside"
      >
        <View
          style={
            {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            editable={true}
            onChangeText={[Function]}
            onEndEditing={[Function]}
            onFocus={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 12,
                  "lineHeight": 18,
                  "marginTop": 8,
                },
                undefined,
                {
                  "backgroundColor": "#FFFFFF",
                  "color": "#1F1F1F",
                },
              ]
            }
            testID="input"
          />
          <View
            style={
              [
                {
                  "position": "absolute",
                  "zIndex": 101,
                },
                undefined,
              ]
            }
            testID="labelContainer"
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onLayout={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Poppins-Regular",
                    },
                    undefined,
                    {
                      "color": undefined,
                      "fontSize": undefined,
                      "left": undefined,
                      "lineHeight": undefined,
                      "top": undefined,
                    },
                    {
                      "color": "#404040",
                    },
                  ]
                }
                testID="inputLabel"
              >
                Label
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 5,
          },
          {
            "alignSelf": "stretch",
            "display": "none",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Poppins-Regular",
              "fontSize": 10,
              "lineHeight": 14,
            },
            undefined,
            {
              "color": "#404040",
            },
          ]
        }
        testID="inputBottomLabel"
      />
    </View>
  </View>
</View>
`;
